src/coordinated_workers/coordinator.py

```diff
@@ -785,6 +791,80 @@ def _reconcile_charm_labels(self) -> None:
             labels=self._coordinated_workers_solution_labels,
         )
 
+    def _get_cluster_scoped_mesh_policy_for_all_cluster_models(
+        self, source_model: str, source_application: str
+    ) -> Dict[str, List[MeshPolicy]]:
+        """Return a mesh policy for a coordinator/worker to allow it to talk to all the other cluster units."""
+        cluster_models = self.cluster.gather_models()
+        mesh_policies: Dict[str, List[MeshPolicy]] = {}
+        for target_model in cluster_models:
+            mesh_policies[target_model] = [
+                MeshPolicy(
+                    source_namespace=source_model,
+                    source_app_name=source_application,
+                    target_namespace=target_model,
+                    target_selector_labels=self._coordinated_workers_solution_labels,
+                    target_type=PolicyTargetType.unit,
+                )
+            ]
+        return mesh_policies
+
+    def _get_cluster_scoped_mesh_policies_by_cluster_model(self) -> Dict[str, List[MeshPolicy]]:
+        """Return all the required cluster internal mesh policies per cluster model."""
+        mesh_policies_by_cluster_model: Dict[str, List[MeshPolicy]] = {}
+        worker_applications: Set[str] = set()
+        # Coordinator -> Everything in the cluster
```

Comment:
```suggestion
        # Everything in the cluster -> Everything in the cluster
```

These policies should grant coordinator->worker* and worker*->worker*

Reply:
Its just its done in two steps. first step `Coordinator -> Everything` and in the next step `Workers -> Everything`